skipper.go

```diff
@@ -760,13 +771,36 @@ func Run(o Options) error {
 		AccessLogDisabled:      o.AccessLogDisabled,
 	}
 
-	if o.EnableBreakers || len(o.BreakerSettings) > 0 {
-		proxyParams.CircuitBreakers = circuit.NewRegistry(o.BreakerSettings...)
+	var theSwarm *swarm.Swarm
+	if o.EnableSwarm {
+		swops := swarm.Options{
+			SwarmPort:        uint16(o.SwarmPort),
+			MaxMessageBuffer: o.SwarmMaxMessageBuffer,
+			LeaveTimeout:     o.SwarmLeaveTimeout,
+			Debug:            log.GetLevel() == log.DebugLevel,
+		}
+		if o.Kubernetes {
+			swops.KubernetesOptions = &swarm.KubernetesOptions{
+				KubernetesInCluster:  o.KubernetesInCluster,
+				KubernetesAPIBaseURL: o.KubernetesURL,
+				Namespace:            o.SwarmKubernetesNamespace,
+				LabelSelectorKey:     o.SwarmKubernetesLabelSelectorKey,
+				LabelSelectorValue:   o.SwarmKubernetesLabelSelectorValue,
+			}
+		}
+		theSwarm, err = swarm.NewSwarm(swops)
+		if err != nil {
+			log.Errorf("failed to init swarm with options %+v: %v", swops, err)
+		}
```

Comment:
shouldn't we call here `defer theSwarm.Leave()`?

Reply:
done